[Bug]: Cannot import testcontainers-go without locking to specific versions of indirect dependencies

### Testcontainers version

0.16.0

### Using the latest Testcontainers version?

Yes

### Host OS

Macos

### Host arch

x86

### Go version

1.19

### Docker version

```shell
Client:
 Cloud integration: v1.0.29
 Version:           20.10.20
 API version:       1.41
 Go version:        go1.18.7
 Git commit:        9fdeb9c
 Built:             Tue Oct 18 18:20:35 2022
 OS/Arch:           darwin/amd64
 Context:           default
 Experimental:      true

Server: Docker Desktop 4.13.1 (90346)
 Engine:
  Version:          20.10.20
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.18.7
  Git commit:       03df974
  Built:            Tue Oct 18 18:18:35 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.8
  GitCommit:        9cd3357b7fd7218e4aec3eae239db1f68a5a6ec6
 runc:
  Version:          1.1.4
  GitCommit:        v1.1.4-0-g5fd4c4d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```


### Docker info

```shell
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc., v0.9.1)
  compose: Docker Compose (Docker Inc., v2.12.1)
  dev: Docker Dev Environments (Docker Inc., v0.0.3)
  extension: Manages Docker extensions (Docker Inc., v0.2.13)
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc., 0.6.0)
  scan: Docker Scan (Docker Inc., v0.21.0)

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 24
 Server Version: 20.10.20
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 9cd3357b7fd7218e4aec3eae239db1f68a5a6ec6
 runc version: v1.1.4-0-g5fd4c4d
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
  cgroupns
 Kernel Version: 5.15.49-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 7.772GiB
 Name: docker-desktop
 ID: UT6I:Z3MH:BVO7:4GHT:YRHP:MGEJ:734U:DELR:HXWX:AEIS:XLYK:VATP
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5000
  127.0.0.0/8
 Live Restore Enabled: false
```


### What happened?

Cannot import testcontainers-go without a bunch of indirect dependencies failing. 

I have to add the following the replace directives to make it work.

```
replace (
	github.com/docker/cli => github.com/docker/cli v20.10.3-0.20221013132413-1d6c6e2367e2+incompatible // 22.06 master branch
	github.com/docker/docker => github.com/docker/docker v20.10.3-0.20221013203545-33ab36d6b304+incompatible // 22.06 branch
	github.com/moby/buildkit => github.com/moby/buildkit v0.10.1-0.20220816171719-55ba9d14360a // same as buildx

	github.com/opencontainers/runc => github.com/opencontainers/runc v1.1.2 // Can be removed on next bump of containerd to > 1.6.4

	// For k8s dependencies, we use a replace directive, to prevent them being
	// upgraded to the version specified in containerd, which is not relevant to the
	// version needed.
	// See https://github.com/docker/buildx/pull/948 for details.
	// https://github.com/docker/buildx/blob/v0.8.1/go.mod#L62-L64
	k8s.io/api => k8s.io/api v0.22.4
	k8s.io/apimachinery => k8s.io/apimachinery v0.22.4
	k8s.io/client-go => k8s.io/client-go v0.22.4
)
```

If I don't lock these indirect dependencies to these versions (and use `latest`) things break.

### Relevant log output

_No response_

### Additional information

_No response_

@mdelapenya I reviewed #629, yes looks like it is related but the solution proposed there isn't sufficient for the errors I am seeing. I tried the suggested `replace` directive you mentioned. While the single replace directive, ```
replace github.com/docker/docker => github.com/docker/docker v20.10.3-0.20221013203545-33ab36d6b304+incompatible // 22.06 branch
```
resolves my `go.mod` issues, however, when I run my tests using `go test ./... -cover`, I get the following errors

```
# github.com/moby/buildkit/session/filesync
../../../go/pkg/mod/github.com/moby/buildkit@v0.10.4/session/filesync/filesync.go:112:20: cannot use dir.Map (variable of type func(string, *types.Stat) bool) as type fsutil.MapFunc in struct literal
# github.com/docker/cli/cli/registry/client
../../../go/pkg/mod/github.com/docker/cli@v20.10.19+incompatible/cli/registry/client/endpoint.go:93:40: assignment mismatch: 3 variables but registry.PingV2Registry returns 2 values
# github.com/docker/buildx/driver/kubernetes/manifest
../../../go/pkg/mod/github.com/docker/buildx@v0.9.1/driver/kubernetes/manifest/manifest.go:92:9: unknown field 'Handler' in struct literal of type "k8s.io/api/core/v1".Probe
../../../go/pkg/mod/github.com/docker/buildx@v0.9.1/driver/kubernetes/manifest/manifest.go:92:25: undefined: v1.Handler
```
The only way I can resolve this is by using **ALL** of the directives from [here](https://github.com/baez90/testcontainers-go/blob/88f3c4b11caa193513f6ad6cef35b47f845bf192/go.mod#L28-L43)

--- a/docs/quickstart/gotest.md
+++ b/docs/quickstart/gotest.md
@@ -20,7 +20,22 @@ go get github.com/testcontainers/testcontainers-go
 	We expect this to be removed in the next releases of _Testcontainers for Go_.
 
 	```
-	replace github.com/docker/docker => github.com/docker/docker v20.10.3-0.20221013203545-33ab36d6b304+incompatible // 22.06 branch
+	replace (
+		github.com/docker/cli => github.com/docker/cli v20.10.3-0.20221013132413-1d6c6e2367e2+incompatible // 22.06 master branch
+		github.com/docker/docker => github.com/docker/docker v20.10.3-0.20221013203545-33ab36d6b304+incompatible // 22.06 branch
+		github.com/moby/buildkit => github.com/moby/buildkit v0.10.1-0.20220816171719-55ba9d14360a // same as buildx
+
+		github.com/opencontainers/runc => github.com/opencontainers/runc v1.1.2 // Can be removed on next bump of containerd to > 1.6.4
+
+		// For k8s dependencies, we use a replace directive, to prevent them being
+		// upgraded to the version specified in containerd, which is not relevant to the
+		// version needed.
+		// See https://github.com/docker/buildx/pull/948 for details.
+		// https://github.com/docker/buildx/blob/v0.8.1/go.mod#L62-L64
+		k8s.io/api => k8s.io/api v0.22.4
+		k8s.io/apimachinery => k8s.io/apimachinery v0.22.4
+		k8s.io/client-go => k8s.io/client-go v0.22.4
+	)
 	```
 
 ## 2. Spin up Redis

--- a/docs/quickstart/gotest.md
+++ b/docs/quickstart/gotest.md
@@ -13,6 +13,15 @@ We use [gomod](https://blog.golang.org/using-go-modules) and you can get it inst
 go get github.com/testcontainers/testcontainers-go
 ```
 
+Given the version includes the Compose dependency, and the Docker folks added [a replace directive until the upcoming Docker 22.06 release is out](https://github.com/docker/compose/issues/9946#issuecomment-1288923912),
+we were forced to add it too, causing consumers of _Testcontainers for Go_ to add the following replace directive to their `go.mod` files.
+
+```
+replace github.com/docker/docker => github.com/docker/docker v20.10.3-0.20221013203545-33ab36d6b304+incompatible // 22.06 branch
+```
+
+We expect this to be removed in the next releases of _Testcontainers for Go_.
+
 ## 2. Spin up Redis
 
 ```go

--- a/docs/quickstart/gotest.md
+++ b/docs/quickstart/gotest.md
@@ -13,14 +13,15 @@ We use [gomod](https://blog.golang.org/using-go-modules) and you can get it inst
 go get github.com/testcontainers/testcontainers-go
 ```
 
-Given the version includes the Compose dependency, and the Docker folks added [a replace directive until the upcoming Docker 22.06 release is out](https://github.com/docker/compose/issues/9946#issuecomment-1288923912),
-we were forced to add it too, causing consumers of _Testcontainers for Go_ to add the following replace directive to their `go.mod` files.
+!!!warning
 
-```
-replace github.com/docker/docker => github.com/docker/docker v20.10.3-0.20221013203545-33ab36d6b304+incompatible // 22.06 branch
-```
+	Given the version includes the Compose dependency, and the Docker folks added [a replace directive until the upcoming Docker 22.06 release is out](https://github.com/docker/compose/issues/9946#issuecomment-1288923912),
+	we were forced to add it too, causing consumers of _Testcontainers for Go_ to add the following replace directive to their `go.mod` files.
+	We expect this to be removed in the next releases of _Testcontainers for Go_.
 
-We expect this to be removed in the next releases of _Testcontainers for Go_.
+	```
+	replace github.com/docker/docker => github.com/docker/docker v20.10.3-0.20221013203545-33ab36d6b304+incompatible // 22.06 branch
+	```
 
 ## 2. Spin up Redis
 

